### PR TITLE
Implementing the unit tests

### PR DESCRIPTION
**Pull request by SergeyGSA**
Added unit tests to:

- error-page component
- message-bar component
- validation-hint component(some unit tests)

### CODE REVIEW

File: src/app/shared/components/message-bar/message-bar.component.spec.ts [hunk 1]
```diff
@@ -1,29 +1,70 @@
...
 import { TranslateModule } from '@ngx-translate/core';
 
 import { MessageBarComponent } from './message-bar.component';
+import { MessageBar } from 'shared/models/messageBar.model';
 
 describe('SnackBarComponent', () => {
   let component: MessageBarComponent;
   let fixture: ComponentFixture<MessageBarComponent>;
+  let matSnackBar: MatSnackBar;
+  let mockData: MessageBar = {
```
Comment (humanmantis) [inline src/app/shared/components/message-bar/message-bar.component.spec.ts]: It is better to set the value inside beforeEach

File: src/app/shared/components/validation-hint/validation-hint.component.spec.ts [hunk 2]
```diff
@@ -23,4 +24,84 @@ describe('ValidationHintComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+  
+  describe('tests of checkInvalidText method', () => {
```
Comment (humanmantis) [inline src/app/shared/components/validation-hint/validation-hint.component.spec.ts]: We can name the group just `checkInvalidText method` or `#checkInvalidText`

File: src/app/shared/components/validation-hint/validation-hint.component.spec.ts [hunk 3]
```diff
@@ -23,4 +24,84 @@ describe('ValidationHintComponent', () => {
...
+    const errorsCasesMap = [
+      ['invalidSymbols', 'NAME_REGEX', NAME_REGEX, true, false, false, false, false],
+      ['invalidCharacters', 'NO_LATIN_REGEX', NO_LATIN_REGEX, false, true, false, false, false],
+      ['invalidStreet', 'STREET_REGEX', STREET_REGEX, false, false, true, false, false],
+      ['invalidHouse', 'HOUSE_REGEX', HOUSE_REGEX, false, false, false, true, false],
+      ['invalidSectionName', 'SECTION_NAME_REGEX', SECTION_NAME_REGEX, false, false, false, false, true],
+    ];
+
+    test.each(errorsCasesMap)('should assign to %s TRUE if required pattern equal to %s',
```
Comment (humanmantis) [inline src/app/shared/components/validation-hint/validation-hint.component.spec.ts]: `test.each` is a good way to test multiple variants, but a more clear way is to create a separate test for each condition

File: src/app/shared/components/validation-hint/validation-hint.component.spec.ts [hunk 4]
```diff
@@ -23,4 +24,84 @@ describe('ValidationHintComponent', () => {
...
+      (_, __, requiredPattern, invalidSymbols, invalidCharacters, invalidStreet, invalidHouse, invalidSectionName) => {
+        const errors = { pattern: { requiredPattern } };
+        component['checkInvalidText'](errors);
```
Comment (humanmantis) [inline src/app/shared/components/validation-hint/validation-hint.component.spec.ts]: Usually, in unit tests, we check only public methods

File: src/app/shared/components/validation-hint/validation-hint.component.spec.ts [hunk 5]
```diff
@@ -23,4 +24,84 @@ describe('ValidationHintComponent', () => {
...
+
+        expect(component.invalidSymbols).toBe(invalidSymbols);
+        expect(component.invalidCharacters).toBe(invalidCharacters);
+        expect(component.invalidStreet).toBe(invalidStreet);
+        expect(component.invalidHouse).toBe(invalidHouse);
+        expect(component.invalidSectionName).toBe(invalidSectionName);
+      });
+  });
+
+  describe('tests of checkMatDatePicker method', () => {
```
Comment (humanmantis) [inline src/app/shared/components/validation-hint/validation-hint.component.spec.ts]: Rename group

**Comment by humanmantis** [comment]
Structure your tests with 3 sections Arrange, Act & Assert. Following this structure ensures that the tests are more understandable.
Add appropriate comments to the tests.
```
it('test', () => {
// arrange
spyOn(...);
...
// act
component.method();
...
// assert
expect(...).toBe(...)
});
```

**Review by humanmantis — CHANGES_REQUESTED** [review]

File: src/app/shared/components/validation-hint/validation-hint.component.spec.ts (hunk 4 above)
Comment (SergeyGSA) [reply src/app/shared/components/validation-hint/validation-hint.component.spec.ts]: So don't we have to write tests of private methods or it depends on situation and time that we have?

Comment (humanmantis) [reply src/app/shared/components/validation-hint/validation-hint.component.spec.ts]: Anyway, we use private methods elsewhere in public/protected methods.
If it requires much more effort, you can test a separate part of the logic that is in a private method.
Testing private methods is okay, but not preferred.

File: src/app/shared/components/message-bar/message-bar.component.spec.ts [hunk 6]
```diff
@@ -1,29 +1,76 @@
...
 import { MessageBarComponent } from './message-bar.component';
+import { MessageBar } from 'shared/models/messageBar.model';
 
 describe('SnackBarComponent', () => {
   let component: MessageBarComponent;
   let fixture: ComponentFixture<MessageBarComponent>;
+  let matSnackBar: MatSnackBar;
+  let mockData: MessageBar;
+
+  beforeEach(() => {
```
Comment (humanmantis) [inline src/app/shared/components/message-bar/message-bar.component.spec.ts]: We already have a `beforeEach` block, you can move it before `TestBed.configureTestingModule` in the same block.

**Review by humanmantis — CHANGES_REQUESTED** [review]

**Review by humanmantis — APPROVED** [review]

**Pull request merged**